### DISCUSSION TITLE
The key is not bound to any value in context RequestContext.

### DISCUSSION THREAD

**Discussion by paramanand-quickship** [loopbackio/loopback-next#8482] [Q&A] [answered]
failed with status code 500. ResolutionError: The key 'controllers.helpers.GrCnHelper' is not bound to any value in context RequestContext-pANyYYgdQCGuSCy_pEAoVg-5 (context: RequestContext-pANyYYgdQCGuSCy_pEAoVg-5, binding: controllers.helpers.GrCnHelper, resolutionPath: controllers.GRcNController --> @GRcNController.constructor[8])

 file structure controllers > helpers > grcn.Helper.ts

**Comment by achrinza** [comment, marked as answer]
Without seeing the code, I'd check if `GrCnHelper` is actually bound to Context at some point. The error essentially means that the framework was not able to resolve that Binding because it was not added.

**Reply by paramanand-quickship** [reply]
I want to call create function from my controller


import {Application, CoreBindings, inject} from '@loopback/core';
import {IsolationLevel, repository, Transaction} from '@loopback/repository';
import {Request, RestBindings} from '@loopback/rest';
import moment from 'moment';
import {AppCode} from '../../models';
import {
  GRcNRepository,

  TxnWiseSuppBalRepository,
} from '../../repositories';
import {DateUtils, ResponseHelper} from '../../utils';
import {CounterController} from '../counter.controller';


export class GrCnHelper {
  constructor(
    @repository(GRcNRepository)
    public gRcNRepository: GRcNRepository,
    )
	{}
	async create(){
	}
	}

**Reply by paramanand-quickship** [reply]
I try to inject GrCnHelper class in my controller but it gives error

**Reply by achrinza** [reply]
Is GrCnHelper ever bound/added to Context? Unlike the Controller (which is a known artifact and is bound automatically by the Booter), GcCnHelper would not be automatically bound to Context. You'll need to bind it manually in `application.ts`:

```
// Inside the Application constructor
this.bind('controllers.helpers.GrCnHelper').toClass(GrCnHelper);
```

(Note that this is one way to bind a class to Context, and there's other approaches available depending on your requirement.)

Manual binding is required because, unlike common artefacts recognised by LoopBack 4, LB4 does not know how to handle custom artefacts (e.g. binding strategy, binding name).

More information can be found in the docs:

- https://loopback.io/doc/en/lb4/Configuring-applications.html
- https://loopback.io/doc/en/lb4/Binding.html

**Comment by paramanand-quickship** [comment]
Thanks 👍 Its working
